From fccf617924a23691fc52a9529ed198e8901b821c Mon Sep 17 00:00:00 2001
From: Chih-Hung Hsieh <chh@google.com>
Date: Mon, 20 Nov 2017 16:02:42 -0800
Subject: [PATCH 5/8] Use -Werror in hardware/interfaces/biometrics/fingerprint

* Fix warning of unused function call result.

Bug: 66996870
Test: build with WITH_TIDY=1
Change-Id: I91da207b80d0715d015195357d645f8cf324fa1e
---
 biometrics/fingerprint/2.0/Android.bp  | 1 +
 biometrics/fingerprint/2.0/service.cpp | 4 +++-
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/biometrics/fingerprint/2.0/Android.bp b/biometrics/fingerprint/2.0/Android.bp
index 86d3fab..372f84e 100644
--- a/biometrics/fingerprint/2.0/Android.bp
+++ b/biometrics/fingerprint/2.0/Android.bp
@@ -15,6 +15,7 @@
 // limitations under the License.
 cc_binary {
     name: "android.hardware.biometrics.fingerprint@2.0-service",
+    defaults: ["hidl_defaults"],
     relative_install_path: "hw",
     init_rc: ["android.hardware.biometrics.fingerprint@2.0-service.rc"],
     srcs: ["service.cpp", "BiometricsFingerprint.cpp"],
diff --git a/biometrics/fingerprint/2.0/service.cpp b/biometrics/fingerprint/2.0/service.cpp
index 1697c07..edfaac4 100644
--- a/biometrics/fingerprint/2.0/service.cpp
+++ b/biometrics/fingerprint/2.0/service.cpp
@@ -35,7 +35,9 @@ int main() {
     configureRpcThreadpool(1, true /*callerWillJoin*/);
 
     if (bio != nullptr) {
-        bio->registerAsService();
+        if (::android::OK != bio->registerAsService()) {
+            return 1;
+        }
     } else {
         ALOGE("Can't create instance of BiometricsFingerprint, nullptr");
     }
-- 
2.17.1

